lib/tree_sitter/mixins/language.rb

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+
+module TreeSitter
+  # Mixins.
+  module Mixins
+    # Language Mixin.
+    module Language
+      # A colon-seprated list of paths pointing to directories that can contain parsers.
+      # Order matters.
+      # Takes precedence over default lookup paths.
+      ENV_PARSERS =
+        ENV['TREE_SITTER_PARSERS']
+          &.split(':')
+          &.map { |v| Pathname(v) }
+          .freeze
+
+      # The default paths we use to lookup parsers when no specific
+      # {TreeStand::Config#parser_path} is `nil`.
+      # Order matters.
+      LIBDIRS = [
+        'tree-sitter-parsers',
+        '/opt/local/lib',
+        '/opt/lib',
+        '/usr/local/lib',
+        '/usr/lib',
+      ].map { |p| Pathname(p) }.freeze
+
+      # The library directories we need to look into.
+      #
+      # @return [Array<Pathname>] the list of candidate places to use when searching for parsers.
+      #
+      # @see ENV_PARSERS
+      # @see LIBDIRS
+      def lib_dirs
+        [
+          *ENV_PARSERS,
+          *(TreeStand.config.parser_path ? [TreeStand.config.parser_path] : LIBDIRS),
+        ].compact
+      end
+
+      # The platform-specific extension of the parser.
+      # @return [String] `dylib` or `so` for mac or linux.
+      def ext
+        case Gem::Platform.local.os
+        in   /darwin/ then 'dylib'
+        else               'so'
+        end
+      end
+
+      # Lookup a parser by name.
+      #
+      # Precedence:
+      # 1. `Env['TREE_SITTER_PARSERS]`
+      # 2. {TreeStand::Config#parser_path}
+      # 3. {LIBDIRS}
+      #
+      # If a {TreeStand::Config#parser_path} is `nil`, {LIBDIRS} is used.
+      # If a {TreeStand::Config#parser_path} is a {::Pathname}, {LIBDIRS} is ignored.
+      def search_for_lib(name)
+        files = [
+          name,
+          "tree-sitter-#{name}",
+          "libtree-sitter-#{name}",
+        ].map { |v| "#{v}.#{ext}" }
+
+        res = lib_dirs
+          .product(files)
+          .find do |dir, so|
+            path = dir / so
+            path = dir / name / so if !path.exist?
+            break path.expand_path if path.exist?
+          end
+        res.is_a?(Array) ? nil : res
```

Comment:
I don't understand this.  find returns `nil` when it finds not.

IMHO you don't need `res`.